UMAA 6

rtidgreenberg opened the pull request:
- Update to UMAA 6 idl
- Create AutoPilot/USVnav components using XML App Creation
- Create example apps using Modern C++ and Python
- Generated Python and C++ helper files for UMAA types
- Create start_component script

angelrti commented:
It is difficult to write some comments in the diff, so I'm adding them here:

* Delete the `examples/build` folder. I can see it is in the .gitignore, but we need to remove it if there are some commits before
* Remove generated files and add an option to generate them from CMake. I have seen that we have generated C++ and Python files under `examples/resources/umaa_6/gen_types` (note, in the Python folder it also includes the `__pycache__` that includes binaries files). Additionally, XML files can be generated as well using CMake. If for convenience you really need the XML files to be in the repository, my recommendation in to have 2 equivalent folders, one of them called `idl` and the other one `xml`
  * Note: here I generate code and also XML files for a different datamodel: https://github.com/rticommunity/dds-datamodels-mdpnp/blob/version/1.5-enhanced/CMakeLists.txt
 * I would organize the repo with the following top level folders:
   * `examples`
     * `examples\srcCxx`
     * `examples\srcPy`
   * `resources` (that contains the scripts directory and other resources)
   * `datamodel\umaa\idl`
   * `rticonnextdds-cmake-utils`
   * `CMakeLists.txt`
     * Then this top-level CMakeLists will call the one in the examples folder
 * c++ example:
   * if the `application.hpp` is generated, we can leave it out of the repo
   * If I'm not missing anything, it looks like the function `process_samples` is not being used
   * In the function `process_keyed_samples` we set the `keyed_data_map`, but I don't see where we use it
 * python example:
   * Try to split the big lines into smaller ones to help readability (ideally 80 chars, but it is ok to use up to 100 chars)
   * In globalVector: add a default value for the `--file`, if that points to a file in the repo folder, we can point to that file by default
   * Add more info about the example itself in argparser description, including the type of entity that we are running (publisher)
   * This is a minor thing, but there are some constant values that we are setting in the main loop, we could do it outside the loop

**Note**: feel free to reach out to me about any of these comments

rtidgreenberg commented:
- examples/build generated from .vscode/settings.json
- Removed C++ generated files, possibly move to external datamodel repo in future for convenience
- Kept Python generated files, no support for PYTHON flag in cmake-utils, created issue
- Changed folder structure to reflect suggestion
- `application.hpp` is not generated, left as is
- `process_samples` not being used, left as reference, added comment for clarity
- `keyed_data_map` is an internal variable for the `process_keyed_samples` function. 
- Added additional line spacing/comments to Python scripts
- Defined constants outside of loop where applicable.
